extraConfigDpuTenant.py

```diff
@@ -141,197 +154,125 @@ def ExtraConfigDpuTenant(cc: ClustersConfig, cfg: ExtraConfigArgs, futures: Dict
+        mgmtPortResourceName = "openshift.io/" + mgmtResourceName
+        logger.info(f"Creating Config Map for mgmt port resource name {mgmtPortResourceName}")
+        with open('./manifests/tenant/hardware-offload-config.yaml.j2') as f:
+            j2_template = jinja2.Template(f.read())
+            rendered = j2_template.render(mgmtPortResourceName=mgmtPortResourceName)
+            logger.info(rendered)
+
+        with open("/tmp/hardware-offload-config.yaml", "w") as outFile:
+            outFile.write(rendered)
+
+        logger.info(tclient.oc("create -f /tmp/hardware-offload-config.yaml"))
+        # DELTA END
+    else:
+        logger.info("creating config map to put ovn-k into dpu host mode")
+        tclient.oc("create -f manifests/tenant/sriovdpuconfigmap.yaml")
+
+    if new_api:
+        # DELTA: We don't create env-override to set management port. https://github.com/ovn-org/ovn-kubernetes/pull/3467
+        pass
+    else:
+        logger.info("setting ovn kube node env-override to set management port")
+        logger.info(os.getcwd())
+        contents = open("manifests/tenant/setenvovnkube.yaml").read()
+
+        assert cfg.mapping is not None
+        for bfmap in cfg.mapping:
+            a: Dict[str, str] = {}
+            mp = re.sub(r'np\d$', '', bf_port)
+            a["OVNKUBE_NODE_MGMT_PORT_NETDEV"] = f"{mp}v0"
+            contents += f"  {bfmap['worker']}: |\n"
+            for k, v in a.items():
+                contents += f"    {k}={v}\n"
+        open("/tmp/1.yaml", "w").write(contents)
+
+        logger.info("Running create")
+        logger.info(tclient.oc("create -f /tmp/1.yaml"))
 
-    logger.info("Running create")
-    logger.info(tclient.oc("create -f /tmp/1.yaml"))
-
-    for nc in cc.workers:
-        cmd = f"label node {nc.name} network.operator.openshift.io/dpu-host="
+    for e in cc.workers:
+        cmd = f"label node {e.name} network.operator.openshift.io/dpu-host="
         logger.info(tclient.oc(cmd))
-        ip = tclient.get_ip(nc.name)
+        ip = tclient.get_ip(e.name)
         if ip is None:
-            logger.error(f"Failed to get ip for node {nc.name}")
+            logger.error(f"Failed to get ip for node {e.name}")
             sys.exit(-1)
         rh = host.RemoteHost(ip)
         rh.ssh_connect("core")
-        # workaround for https://issues.redhat.com/browse/NHE-335
+
+        if new_api:
+            # DELTA: "ovn-k8s-mp0" port is deleted by OVN-K. https://github.com/ovn-org/ovn-kubernetes/pull/3571
+            result = rh.run("sudo ovs-vsctl show")
+            if "ovn-k8s-mp0" in result.out:
+                logger.info(result.out)
+                logger.info("Unexpected ovn-k8s-mp0 interface found in br-int.")
+                # FIXME: The above patch did not seem to entirely work. We will need to investigate further
+                # For now we will delete the port.
+        else:
+            # workaround for https://issues.redhat.com/browse/NHE-335
```

Comment:
Redundant, this comment can just be applied to the ovs-vsctl del-port command following this if-else.